### PR TITLE
Fixes origami folding techniques not being retained when cloned

### LINKED ISSUE #42331: Oragami Folding Not Retained on Death
So uh took oragami as a meme. Got murdered in the halls after shredding the book in the recycler.
Got cloned. Didn't keep my paper folding skills and sent to perma. Is this working as intended? I thought all the stuff you learned from books was kept when you got cloned.

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
fix: Learning how to construct lethal paper planes is no longer
forgotten on death.
tweak: Orgami folding is always on, and no longer has an action button.
fix: Fixes orgami kits not containing pre-folded lethal paperplanes.
/:cl:

Granter books were reusing a lot of the same code in order to do the
page by page experience, so I unified that.

The description for the origami kit says it comes with paper planes, but
it just came with paper, so I changed that as well.

Otherwise, just code and refactoring.

Fixes #42331.

### CODE REVIEW

File: code/game/objects/items/granters.dm [hunk 1]
```diff
@@ -18,91 +19,90 @@
...
-			to_chat(user, "<span class='notice'>You already know all about [actionname].</span>")
-			qdel(G)
-			return
-	if(used == TRUE && oneuse == TRUE)
-		recoil(user)
+			to_chat(user, "<span class='notice'>You already know all about [topic].</span>")
+			. = FALSE
+	qdel(G)
+
+/obj/item/book/granter/action/onlearned(mob/user)
```
Comment (Cyberboss) [inline code/game/objects/items/granters.dm]:
```suggestion
/obj/item/book/granter/action/onlearned(mob/living/user)
```

File: code/game/objects/items/granters.dm [hunk 2]
```diff
@@ -18,91 +19,90 @@
...
+/obj/item/book/granter/trait
+	var/trait
+
+/obj/item/book/granter/trait/can_learn_check(mob/user)
+	. = TRUE
+	if(user.has_trait(trait, GRANTER_BOOK_TRAIT))
+		to_chat(user, "<span class='notice'>You already know all about [topic].</span>")
+		. = FALSE
+
+/obj/item/book/granter/trait/onlearned(mob/user)
```
Comment (Cyberboss) [inline code/game/objects/items/granters.dm]:
```suggestion
/obj/item/book/granter/trait/onlearned(mob/living/user)
```

File: code/game/objects/items/granters.dm [hunk 3]
```diff
@@ -18,91 +19,90 @@
...
 	var/spell
-	var/spellname = "conjure bugs"
 
-/obj/item/book/granter/spell/attack_self(mob/user)
-	. = ..()
-	if(!.)
-		return
-	if(!spell)
-		return
+/obj/item/book/granter/spell/can_learn_check(mob/user)
```
Comment (Cyberboss) [inline code/game/objects/items/granters.dm]:
```suggestion
/obj/item/book/granter/spell/can_learn_check(mob/living/user)
```

File: code/game/objects/items/granters.dm [hunk 4]
```diff
@@ -111,36 +111,23 @@
...
 /obj/item/book/granter/spell/onlearned(mob/user)
-	..()
+	var/obj/effect/proc_holder/spell/S = new spell
+	user.mind.AddSpell(S)
+	user.log_message("learned the spell [S]", LOG_ATTACK, color="orange")
+
 	if(oneuse)
 		user.visible_message("<span class='caution'>[src] glows dark for a second!</span>")
 
+/obj/item/book/granter/spell/recoil(mob/user)
```
Comment (Cyberboss) [inline code/game/objects/items/granters.dm]:
```suggestion
/obj/item/book/granter/spell/recoil(mob/living/user)
```

File: code/game/objects/items/granters.dm [hunk 5]
```diff
@@ -305,41 +292,28 @@
...
-			onlearned(user)
-		reading = FALSE
+		to_chat(user,"<span class='warning'>You already know [topic]!</span>")
+		. = FALSE
+	qdel(MA)
+
+/obj/item/book/granter/martial/onlearned(mob/user)
+	var/datum/martial_art/MA = new martial
+
+	to_chat(user, "[greet]")
```
Comment (Cyberboss) [inline code/game/objects/items/granters.dm]:
```suggestion
	to_chat(user, greet)
```

File: code/game/objects/items/granters.dm [hunk 6]
```diff
@@ -305,41 +292,28 @@
...
+	topic = "bug jitsu"
 	var/greet = "You feel like you have mastered the art in breaking code. Nice work, jackass."
 
-/obj/item/book/granter/martial/attack_self(mob/user)
-	. = ..()
-	if(!.)
-		return
-	if(!martial)
-		return
+/obj/item/book/granter/martial/can_learn_check(mob/user)
```
Comment (Cyberboss) [inline code/game/objects/items/granters.dm]:
```suggestion
/obj/item/book/granter/martial/can_learn_check(mob/living/user)
```

File: code/game/objects/items/granters.dm [hunk 7]
```diff
@@ -305,41 +292,28 @@
...
-			to_chat(user, "[greet]")
-			MA.teach(user)
-			user.log_message("learned the martial art [martialname] ([MA])", LOG_ATTACK, color="orange")
-			onlearned(user)
-		reading = FALSE
+		to_chat(user,"<span class='warning'>You already know [topic]!</span>")
+		. = FALSE
+	qdel(MA)
+
+/obj/item/book/granter/martial/onlearned(mob/user)
```
Comment (Cyberboss) [inline code/game/objects/items/granters.dm]:
```suggestion
/obj/item/book/granter/martial/onlearned(mob/living/user)
```

File: code/game/objects/items/granters.dm [hunk 8]
```diff
@@ -18,91 +19,90 @@
...
+	qdel(G)
+
+/obj/item/book/granter/action/onlearned(mob/user)
+	var/datum/action/G = new granted_action
+	G.Grant(user)
+
+/obj/item/book/granter/trait
+	var/trait
+
+/obj/item/book/granter/trait/can_learn_check(mob/user)
```
Comment (Cyberboss) [inline code/game/objects/items/granters.dm]:
```suggestion
/obj/item/book/granter/trait/can_learn_check(mob/living/user)
```

File: code/game/objects/items/granters.dm [hunk 9]
```diff
@@ -18,91 +19,90 @@
...
 	var/granted_action
-	var/actionname = "catching bugs" //might not seem needed but this makes it so you can safely name action buttons toggle this or that without it fucking up the granter, also caps
 
-/obj/item/book/granter/action/attack_self(mob/user)
-	. = ..()
-	if(!.)
-		return
-	if(!granted_action)
-		return
+/obj/item/book/granter/action/can_learn_check(mob/user)
```
Comment (Cyberboss) [inline code/game/objects/items/granters.dm]:
```suggestion
/obj/item/book/granter/action/can_learn_check(mob/living/user)
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by PKPenguin321** [comment]
Not being toggleable is a balance change, since it gives a hard tell against you

**Comment by Cyberboss** [comment]
Your PR is getting old. Either Travis CI is failing, it is still labelled work in progress (change this by adding `[READY]` to the start of your title), or there are unaddressed change requests. This may be closed soon to free up space on our tracker if it remains idle. In the event that it is, please ask in #coderbus on Rizon for it to be reopened once you have resumed work on it.

**Comment by coiax** [comment]
I haven't forgotten about this.

**Comment by ShizCalev** [comment]
Gotta be toggleable :(

File: code/game/objects/items/granters.dm [hunk 10]
```diff
@@ -4,105 +4,105 @@
...
 		to_chat(user, "<span class='warning'>You're already reading this!</span>")
-		return FALSE
-	if(!user.can_read(src))
-		return FALSE
-	return TRUE
+		return
+	if(!user.can_read(src) || !can_learn_check(user))
+		return
+
+	if(used == TRUE && oneuse == TRUE)
```
Comment (Cyberboss) [inline code/game/objects/items/granters.dm]:
```suggestion
	if(used && oneuse)
```

File: code/game/objects/items/granters.dm [hunk 11]
```diff
@@ -4,105 +4,105 @@
...
 	for(var/datum/action/A in user.actions)
 		if(A.type == G.type)
-			to_chat(user, "<span class='notice'>You already know all about [actionname].</span>")
-			qdel(G)
-			return
-	if(used == TRUE && oneuse == TRUE)
-		recoil(user)
+			to_chat(user, "<span class='notice'>You already know all about [topic].</span>")
+			. = FALSE
+	qdel(G)
```
Comment (Cyberboss) [inline code/game/objects/items/granters.dm]: Needs a `break`

File: code/game/objects/items/granters.dm [hunk 12]
```diff
@@ -111,63 +111,50 @@
...
-		if(do_after(user,50, user))
-			to_chat(user, "<span class='notice'>You feel like you've experienced enough to cast [spellname]!</span>")
-			user.mind.AddSpell(S)
-			user.log_message("learned the spell [spellname] ([S])", LOG_ATTACK, color="orange")
-			onlearned(user)
-		reading = FALSE
-
-/obj/item/book/granter/spell/recoil(mob/user)
-	user.visible_message("<span class='warning'>[src] glows in a black light!</span>")
+			. = FALSE
```
Comment (Cyberboss) [inline code/game/objects/items/granters.dm]: Here too

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by Cyberboss** [comment]
Your PR is getting old. Either Travis CI is failing, it is still labelled work in progress (change this by adding `[READY]` to the start of your title), or there are unaddressed change requests. This may be closed soon to free up space on our tracker if it remains idle. In the event that it is, please ask in #coderbus on Rizon for it to be reopened once you have resumed work on it.

**Pull request closed without merging**